lib/get-config.js

```diff
@@ -322,10 +342,28 @@ module.exports = function(options) {
     processExtends(config, options);
     processIgnores(config);
 
-    validateRules(config, options);
+    validateRules(config.rules, config.loadedRules, options);
+
+    if (config.overrides) {
+      for (const override of config.overrides) {
+        // If there are keys in the object which are not yet supported by overrides functionality, throw an error.
+        const overrideKeys = Object.keys(override);
+        const containsValidKeys = overrideKeys.every(item =>
+          SUPPORTED_OVERRIDE_KEYS.includes(item)
+        );
+        if (!containsValidKeys) {
+          throw new Error(
+            'Using `overrides` in `.template-lintrc.js` only supports `files` and `rules` sections. Please update your configuration.'
+          );
+        }
+        validateRules(override.rules, config.loadedRules, options);
+      }
+    }
```

Comment:
Let's extract this to a `validateOverrides` utility method. I'd like to keep `getProjectConfig` "simple" by having it just call a bunch of helper methods (makes it so you don't have to think about how all of the things work).